types/parse/index.d.ts

```diff
@@ -79,6 +80,11 @@ declare enum ErrorCode {
 
 declare namespace Parse {
 
+    // Helper types for extracting an extended generic type from an
+    // (Parse.)Object or Array.
+    type Unarray<T> = T extends Array<infer U> ? U : T;
+    type Unobject<X> = X extends Object<infer U> ? U extends any ? any : U : X;
```

Comment:
Anything `extends` `any`, doesn't it? So, if `X extends Object` this would return `any`. That's why that (following) test passed. The problem isn't solved yet :confused: 

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/f1a13f991a2d4abab3f8fd3e0a3082160af85743/types/parse/parse-tests.ts#L665-L666

Reply:
Ah you're right, obviously. 

The `new Parse.Query(Parse.Object)` is giving me some real headaches... I've pushed a new draft which adds an `UnpackKeys` type to  hopefully solves these 'never' and 'unknown' cases.